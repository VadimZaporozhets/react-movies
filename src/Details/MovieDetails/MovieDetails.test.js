import React from 'react';
import { MovieDetailsComponent } from './index';
import { shallow } from 'enzyme';
import emptyImageUrl from './../../constants';

describe('<MovieDetailsComponent />', () => {
    it('should render movie details component when there is no error', () => {
        const props = {
            title: 'Movie',
            releaseYear: '2006',
            poster_path: emptyImageUrl,
            genres: ['Action', 'Comedy'],
            vote_average: 5,
            overview: 'Test overview',
            classes: expect.any(Object),
            error: ''
        };

        const wrapper = shallow(<MovieDetailsComponent {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render error when it appears', () => {
        const props = {
            error: 'Error',
            classes: expect.any(Object)
        };

        const wrapper = shallow(<MovieDetailsComponent {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
