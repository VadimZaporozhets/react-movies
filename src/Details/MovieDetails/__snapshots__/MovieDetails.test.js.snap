// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetailsComponent /> should render error when it appears 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={40}
>
  <WithStyles(Typography)>
    Error
  </WithStyles(Typography)>
</WithStyles(Grid)>
`;

exports[`<MovieDetailsComponent /> should render movie details component when there is no error 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={40}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Paper)
      elevation={1}
    >
      <WithStyles(Grid)
        container={true}
        spacing={40}
      >
        <WithStyles(Grid)
          item={true}
          md={4}
          xs={12}
        >
          <img
            alt="Movie"
            src="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22288%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20288%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_164edaf95ee%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A14pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_164edaf95ee%22%3E%3Crect%20width%3D%22288%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%2296.32500076293945%22%20y%3D%22118.8%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          alignContent="flex-start"
          container={true}
          item={true}
          md={8}
          spacing={40}
          xs={12}
        >
          <WithStyles(Grid)
            alignItems="center"
            container={true}
            item={true}
            md={10}
            xs={12}
          >
            <WithStyles(Typography)
              className="movie-title"
              variant="h4"
            >
              Movie
            </WithStyles(Typography)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            alignItems="center"
            container={true}
            item={true}
            justify="flex-end"
            md={2}
            xs={12}
          >
            <WithStyles(Badge)
              badgeContent={5}
              color="primary"
            >
              <WithStyles(Typography)
                className="movie-rating"
                variant="button"
              >
                Rating
              </WithStyles(Typography)>
            </WithStyles(Badge)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            container={true}
            item={true}
            spacing={40}
            xs={12}
          >
            <WithStyles(Grid)
              item={true}
            >
              <WithStyles(Typography)
                className="release-year"
                variant="button"
              >
                2006
              </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              <WithStyles(Typography)
                inline={true}
                key="Action"
                variant="button"
              >
                Action
              </WithStyles(Typography)>
              <WithStyles(Typography)
                inline={true}
                key="Comedy"
                variant="button"
              >
                Comedy
              </WithStyles(Typography)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(Typography)
              className="overview"
            >
              Test overview
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(Paper)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
